### PR TITLE
Easier renewable sponge farm in 1.14+

### PR DESCRIPTION
**Pull request by Nameless-123**
There is no rng manipulation as far as I know, so sponge farms are near impossible to make useful
By:Ghoulboy

### CODE REVIEW

**Comment by gnembon** [comment]
I assume you would want that instead of carpet rule for renewableSponges and without messing up with datapacks, which is fine.

File: programs/survival/easier_renewable_sponge.sc [hunk 1]
```diff
@@ -0,0 +1,29 @@
+__on_player_attacks_entity(player,entity)->(
+	schedule(0,'__add_dead_fish',entity)//if not the fish's health isn't 0 and I cant check if its dead or not
+);
+global_puff=0;
+global_trop=0;
+global_salmon=0;
+global_cod=0;
+
+__add_dead_fish(entity)->(
```
Comment (gnembon) [inline programs/survival/easier_renewable_sponge.sc]: global_sponge_fishez = m('Pufferfish', 'Tropical Fish', ...);
..
if (has(global_sponge_fishez, str(entity)) && entity ~ 'health' == 0, ...)

File: programs/survival/easier_renewable_sponge.sc [hunk 2]
```diff
@@ -0,0 +1,29 @@
+__on_player_attacks_entity(player,entity)->(
+	schedule(0,'__add_dead_fish',entity)//if not the fish's health isn't 0 and I cant check if its dead or not
```
Comment (gnembon) [inline programs/survival/easier_renewable_sponge.sc]: I woudn't schedule call for every player kill - also check the entity type here

File: programs/survival/easier_renewable_sponge.sc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	if (entity=='Tropical Fish'&& query(entity,'health')==0,
+		global_trop+=1;
+	);
+	if (entity=='Salmon'&& query(entity,'health')==0,
+		global_salmon+=1;
+	);
+	if (entity=='Cod'&& query(entity,'health')==0,
+		global_cod+=1;
+	);
+	if(global_puff>0 && global_trop >0 && global_salmon>0 && global_cod>0 &&rand(50)==0,
```
Comment (gnembon) [inline programs/survival/easier_renewable_sponge.sc]: I assume you will start getting spongs when you kill at least one of each?
if you would keep player stats in a map, then  you could use all(...) function. But that would reset with every restart - I assume that's intended

File: programs/survival/easier_renewable_sponge.sc (hunk 2 above)
Comment (Nameless-123) [reply programs/survival/easier_renewable_sponge.sc]: ok, i'll fix that

File: programs/survival/easier_renewable_sponge.sc (hunk 3 above)
Comment (Nameless-123) [reply programs/survival/easier_renewable_sponge.sc]: ok yeah, I have no clue clue how to use any form of map. I always end up with northe pointing wst and south pointing east

File: programs/survival/easier_renewable_sponge.sc (hunk 1 above)
Comment (Nameless-123) [reply programs/survival/easier_renewable_sponge.sc]: yeah, i totally understand...

File: programs/survival/easier_renewable_sponge.sc (hunk 3 above)
Comment (gnembon) [reply programs/survival/easier_renewable_sponge.sc]: in this case map or list, doesn't matter that much, but either would look much better than several if's next to each other.

Comment (Nameless-123) [reply programs/survival/easier_renewable_sponge.sc]: ok, yeah, i have no clue how to do this wth lists either

File: programs/survival/easier_renewable_sponge.sc (hunk 1 above)
Comment (Nameless-123) [reply programs/survival/easier_renewable_sponge.sc]: I dont know how to do the rest so ill just do it for the more obvious section

**Comment by gnembon** [comment]
hmm, seems like I forgot to mention - please replace run(`/summon, with spawn(...) and we can merge like that.

**Comment by Nameless-123** [comment]
> hmm, seems like I forgot to mention - please replace run(`/summon, with spawn(...) and we can merge like that.

It never seems to work when I try, so I can't seem to do it

**Pull request merged**